src/class/impl_.rs

```diff
@@ -31,7 +31,19 @@ impl<T> Copy for PyClassImplCollector<T> {}
 ///
 /// Users are discouraged from implementing this trait manually; it is a PyO3 implementation detail
 /// and may be changed at any time.
-pub trait PyClassImpl: Sized {
+pub trait PyClassImpl: Sized + PyTypeInfo {
```

Comment:
I'm not sure this reduces the complexity of our (somewhat complex) trait hierarchy...

Reply:
I agree it's sad, this is because `PyClassAlloc` wanted `PyTypeInfo`. Maybe I can do some other adjusting.